service.subtitles.subztv.gr/service.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+
+'''
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+'''
+
+
+import xbmc
+import urllib,urlparse,re,os,time
+
+from resources.lib import subztvgr
+from lamlib import control
+from lamlib import workers
+
+
+class search:
+    def __init__(self):
+        self.list = []
+
+    def run(self, query=None):
+
+        if not 'Greek' in str(langs).split(','):
+            control.directory(int(sys.argv[1]))
+            control.infoDialog(control.lang(32002).encode('utf-8'))
+            return
+
+
+        if query == None:
+            title = control.infoLabel('VideoPlayer.Title')
+
+            if not re.search(r'[^\x00-\x7F]+', title) == None:
+                title = control.infoLabel('VideoPlayer.OriginalTitle')
+
+            year = control.infoLabel('VideoPlayer.Year')
+
+            tvshowtitle = control.infoLabel('VideoPlayer.TVshowtitle')
+
+            season = control.infoLabel('VideoPlayer.Season')
+
+            episode = control.infoLabel('VideoPlayer.Episode')
+
+            if 's' in episode.lower():
+                season, episode = '0', episode[-1:]
+
+            if not tvshowtitle == '': # episode
+                query = '%s S%02dE%02d' % (tvshowtitle, int(season), int(episode))
+            elif not year == '': # movie
+                query = '%s (%s)' % (title, year)
+            else: # file
+                query, year = xbmc.getCleanMovieTitle(title)
+                if not year == '': query = '%s (%s)' % (query, year)
+
+        self.query = query
+
+
+        threads = []
+
+        threads.append(workers.Thread(self.subztvgr))
+
+        [i.start() for i in threads]
+
+        for i in range(0, 10 * 2):
+            try:
+                is_alive = [x.is_alive() for x in threads]
+                if all(x == False for x in is_alive): break
+                time.sleep(0.5)
```

Comment:
Added it in my new PR
https://github.com/lambda81/service.subtitles.subztv.gr/blob/master/service.py#L77